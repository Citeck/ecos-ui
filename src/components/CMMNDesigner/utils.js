import ModelUtil from 'cmmn-js/lib/util/ModelUtil';

export const GROUP_CUSTOM = 'custom';

function getType(element) {
  const definition = ModelUtil.getDefinition(element);
  return definition && definition.$type;
}

function getEcosType(element) {
  const definition = ModelUtil.getDefinition(element);
  return definition && definition.get ? definition.get('ecos:cmmnType') : '';
}

const initialDiagram = `<?xml version="1.0" encoding="UTF-8"?>
<cmmn:definitions
xmlns:dc="http://www.omg.org/spec/CMMN/20151109/DC"
xmlns:cmmndi="http://www.omg.org/spec/CMMN/20151109/CMMNDI"
xmlns:cmmn="http://www.omg.org/spec/CMMN/20151109/MODEL"
xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
xmlns:ecos="http://www.citeck.ru/ecos/cmmn"
id="Definitions_0uso40t" targetNamespace="http://bpmn.io/schema/cmmn" exporter="cmmn-js (https://demo.bpmn.io/cmmn)" exporterVersion="0.20.0">
  <cmmn:case id="Case_0aitqmd">
    <cmmn:casePlanModel id="CasePlanModel_1w94rbv" name="A CasePlanModel" />
  </cmmn:case>
  <cmmn:case id="Case_00b21zh">
    <cmmn:casePlanModel id="CasePlanModel_0dl1sfc">
      <cmmn:planItem id="PlanItem_1hiokxy" definitionRef="Task_1can4r7" />
      <cmmn:task id="Task_1can4r7" name="Action" ecos:cmmnType="Action" />
    </cmmn:casePlanModel>
  </cmmn:case>
  <cmmndi:CMMNDI>
    <cmmndi:CMMNDiagram id="CMMNDiagram_1">
      <cmmndi:Size width="500" height="500" />
      <cmmndi:CMMNShape id="CasePlanModel_0dl1sfc_di" cmmnElementRef="CasePlanModel_0dl1sfc">
        <dc:Bounds x="264" y="331" width="400" height="250" />
        <cmmndi:CMMNLabel />
      </cmmndi:CMMNShape>
      <cmmndi:CMMNShape id="PlanItem_1hiokxy_di" cmmnElementRef="PlanItem_1hiokxy">
        <dc:Bounds x="375" y="415" width="100" height="80" />
        <cmmndi:CMMNLabel />
      </cmmndi:CMMNShape>
    </cmmndi:CMMNDiagram>
  </cmmndi:CMMNDI>
</cmmn:definitions>
`;

export default {
  initialDiagram,
  getEcosType,
  getType
};
