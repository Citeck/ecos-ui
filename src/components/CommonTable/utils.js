import get from 'lodash/get';

import { JOURNAL_MIN_HEIGHT, JOURNAL_MIN_HEIGHT_MOB } from '../Journals/constants';

export const getMinHeight = isMobile => (isMobile ? JOURNAL_MIN_HEIGHT_MOB : JOURNAL_MIN_HEIGHT);

export const getMaxHeight = (isMobile, _headerRef, _footerRef) => {
  const headH = (_headerRef && get(_headerRef.getBoundingClientRect?.(), 'bottom')) || 0;
  const jFooterH = (_footerRef && get(_footerRef, 'offsetHeight')) || 0;
  const footerH = get(document.querySelector('.app-footer'), 'offsetHeight') || 0;
  const scrollHeight = get(document.querySelector('.ecos-kanban__scroll_h'), 'offsetHeight') || 0;
  const height = document.documentElement.clientHeight - headH - jFooterH - footerH - scrollHeight - 48;

  const minHeight = getMinHeight(isMobile);
  return Math.max(height, minHeight);
};
