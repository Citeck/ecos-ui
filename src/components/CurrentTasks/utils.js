import PropTypes from 'prop-types';
import { t } from '../../helpers/util';
import { DataFormatTypes } from '../../constants';

export const CurrentTaskPropTypes = {
  id: PropTypes.string,
  title: PropTypes.string,
  actors: PropTypes.string,
  deadline: PropTypes.any
};

export const DisplayedColumns = {
  title: {
    key: 'title',
    label: t('Задача'),
    order: 0
  },
  actors: {
    key: 'actors',
    label: t('Исполнитель'),
    order: 1
  },
  deadline: {
    key: 'deadline',
    label: t('Срок до'),
    order: 2,
    format: DataFormatTypes.DATE
  }
};
