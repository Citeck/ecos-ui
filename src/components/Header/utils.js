import { isEmpty } from 'lodash';

export function getIconClassMenu(id, specialClass) {
  switch (id) {
    case 'HEADER_USER_MENU_MY_PROFILE':
      return 'icon-User_avatar';
    case 'HEADER_USER_MENU_AVAILABILITY':
      return specialClass;
    case 'HEADER_USER_MENU_PASSWORD':
      return '';
    case 'HEADER_USER_MENU_FEEDBACK':
      return 'icon-notify-dialogue';
    case 'HEADER_USER_MENU_REPORTISSUE':
      return 'icon-big_alert';
    case 'HEADER_USER_MENU_LOGOUT':
      return 'icon-exit';
    case 'HEADER_SITE_INVITE':
      return '';
    case 'HEADER_CUSTOMIZE_SITE_DASHBOARD':
      return '';
    case 'HEADER_EDIT_SITE_DETAILS':
      return '';
    case 'HEADER_CUSTOMIZE_SITE':
      return '';
    case 'HEADER_LEAVE_SITE':
      return '';
    case 'HEADER_SITE_JOURNALS':
      return '';
    default:
      return '';
  }
}

export function getSpecialClassByState(id, params = {}) {
  if (!isEmpty(params)) {
    const colorOn = 'icon_on';
    const colorOff = 'icon_off';

    switch (id) {
      case 'HEADER_USER_MENU_AVAILABILITY':
        return params.available ? `icon-User_avatar_on ${colorOn}` : `icon-User_avatar_off ${colorOff}`;
      default:
        return false;
    }
  }

  return false;
}
