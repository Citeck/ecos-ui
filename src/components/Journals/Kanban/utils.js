export function isDropDisabled({ readOnly, isLoadingCol, columnInfo }) {
  return readOnly || isLoadingCol || columnInfo.id === 'EMPTY';
}
