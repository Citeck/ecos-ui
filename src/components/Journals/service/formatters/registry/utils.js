import isFunction from 'lodash/isFunction';
import { IS_TEST_ENV } from '../../../../../helpers/util';
import BaseFormatter from '../BaseFormatter';

export const getAllFormattersModules = requireContextFunc => {
  return requireContextFunc
    .keys()
    .map(requireContextFunc)
    .map(FormatterModule => FormatterModule.default)
    .filter(
      FormatterModule => Boolean(FormatterModule) && isFunction(FormatterModule) && FormatterModule.prototype instanceof BaseFormatter
    );
};

export const requireContextNodeJS = (base, scanSubDirectories, regularExpression) => {
  if (!IS_TEST_ENV) {
    throw new Error("You should't call function requireContextNodeJS outside of NodeJS environmvent");
  }

  const fs = require('fs');
  const path = require('path');
  const files = {};

  function readDirectory(directory) {
    fs.readdirSync(directory).forEach(file => {
      const fullPath = path.resolve(directory, file);

      if (fs.statSync(fullPath).isDirectory()) {
        if (scanSubDirectories) readDirectory(fullPath);

        return;
      }

      if (!regularExpression.test(fullPath)) return;

      files[fullPath] = true;
    });
  }
  readDirectory(path.resolve(__dirname, base));

  function Module(file) {
    // IS_TEST_ENV is not used because of eslint
    if (process.env.NODE_ENV === 'test') {
      return require(file);
    }

    return;
  }

  Module.keys = () => Object.keys(files);

  return Module;
};
