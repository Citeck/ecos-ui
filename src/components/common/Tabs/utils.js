import PropTypes from 'prop-types';

export const commonOneTabPropTypes = {
  id: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  label: PropTypes.string,
  className: PropTypes.string,
  isActive: PropTypes.bool,
  hasHover: PropTypes.bool,
  onClick: PropTypes.func
};

export const commonOneTabDefaultProps = {
  label: '',
  className: '',
  hasHover: false,
  onClick: () => null
};

export const commonTabsPropTypes = {
  items: PropTypes.arrayOf(
    PropTypes.shape({
      ...commonOneTabPropTypes
    })
  ),
  className: PropTypes.string,
  hasHover: PropTypes.bool
};

export const commonTabsDefaultProps = {
  items: [],
  className: '',
  hasHover: false
};
