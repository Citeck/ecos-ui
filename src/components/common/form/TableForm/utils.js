export const getAllComponents = components => {
  const getComponents = component => {
    if (component.columns) {
      return getAllComponents(component.columns);
    }

    if (component.components) {
      return getAllComponents(component.components);
    }

    return component;
  };

  return components.reduce((result, component) => result.concat(getComponents(component)), []);
};
