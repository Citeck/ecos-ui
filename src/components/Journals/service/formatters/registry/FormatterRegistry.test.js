import { getAllFormattersModules, requireContextNodeJS } from './utils';

describe('FormatterRegistry', () => {
  it('should skip not extended from BaseModule', () => {
    const modules = requireContextNodeJS('../__mocks__/skipUtils', true, /Formatter.js$/);
    const formatterModules = getAllFormattersModules(modules);
    expect(formatterModules.length).toEqual(0);
  });

  it('should add formatter from folder and not', () => {
    const modules = requireContextNodeJS('../__mocks__', true, /Formatter.js$/);
    const formatterModules = getAllFormattersModules(modules);
    expect(formatterModules.length).toEqual(2);
    expect(formatterModules).toMatchSnapshot();
  });
});
