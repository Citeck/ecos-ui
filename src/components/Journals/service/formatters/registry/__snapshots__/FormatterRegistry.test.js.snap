// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormatterRegistry should add formatter from folder and not 1`] = `
Array [
  [Function],
  [Function],
]
`;
